<launch>
  
  <group ns="pose_ekf">
    <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
        <param name="output_frame" value="odom_combined"/>
        <param name="freq" value="30.0"/>
        <param name="sensor_timeout" value="1.0"/>
        <param name="odom_used" value="true"/>
        <param name="imu_used" value="true"/>
        <param name="vo_used" value="false"/>
        <param name="debug" value="false"/>
        <param name="self_diagnose" value="false"/>
        <remap from="/imu_data" to="/imu"/>
        <remap from="odom" to="/pepper_robot/odom" />
        <param name="base_footprint_frame" value="base_link"/>

    </node>
  </group>

  <group ns="camera_conversion">
    <node pkg="nodelet" type="nodelet" args="manager" name="depth_image_nodelet_manager" output="screen"/>
    <node pkg="nodelet" type="nodelet" name="pepper_pointcloud" args="load depth_image_proc/point_cloud_xyz depth_image_nodelet_manager --no-bond">
        <remap from="camera_info" to="/pepper_robot/camera/depth/depth/camera_info"/>
        <remap from="image_rect" to="/pepper_robot/camera/depth/depth/image_raw"/>
    </node>

    <!-- Pointcloud -> Laserscan -->
     <node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pepper_fake_laser">
        <remap from="cloud_in" to="points" />
        <param name="scan_time" type="double" value="0.2" />
        <param name="range_min" type="double" value="0.4" />
        <param name="range_max" type="double" value="8.0" />
        <param name="use_inf" type="boolean" value="true" />
        <param name="min_height" type="double" value="-0.3" />
        <param name="max_height" type="double" value="0.3" />
        <param name="target_frame" type="string" value="pepper_robot/base_link" />
        <param name="concurrency_level" type="int" value="0" />
    </node>
  </group>

  <group ns="odom_conversion">
    <node pkg="odometry_conversion" type="pose_conversion.py" name="pose_conversion">
        <remap from="ekf_odom_combined" to="/pose_ekf/robot_pose_ekf/odom_combined"/>
        <remap from="combined_odometry" to="/combined_odometry"/>
    </node>
  </group>



</launch>